packages/react/src/components/NumberField/NumberField.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { useLocale } from '@react-aria/i18n';
+import { useNumberField } from '@react-aria/numberfield';
+import { mergeProps, mergeRefs } from '@react-aria/utils';
+import { useNumberFieldState } from '@react-stately/numberfield';
+import { cx } from '@project44-manifest/react-styles';
+import { As, createComponent, Props } from '../../system';
+import {
+  NumberFieldBase,
+  NumberFieldBaseElement,
+  NumberFieldBaseOptions,
+} from '../NumberFieldBase';
+import { useStyles } from './NumberField.styles';
+
+export interface NumberFieldOptions<T extends As = NumberFieldBaseElement>
+  extends NumberFieldBaseOptions<T> {
+  /**
+   * Use custom locals (en-US, fr-FR, de-DE...)
+   */
+  customLocale?: string;
+  /**
+   * The default value (uncontrolled).
+   */
+  defaultValue?: number;
+  /**
+   * Formating options (Intl - ECMAScript Internationalization API)
+   */
+  formatOptions?: Intl.NumberFormatOptions;
+  /**
+   * Temporary Number that occupies the Number input when it is empty.
+   */
+  placeholder?: string;
+  /**
+   * The size of the combobox
+   *
+   * @default 'medium'
+   */
+  size?: 'medium' | 'small';
+  /**
+   * The current value (controlled).
+   */
+  value?: number;
```

Comment:
I'm not going to block this PR for this but FYI the typical architecture would be to utilize `children` prop to pass the value, not prop-driven `value`.

Reply:
I based `NumberField` on existing `TextField` component. So I kept the pattern from it:
https://github.com/project44/manifest/blob/0e6786966315df34897142228a703ba95f503316/packages/react/src/components/TextField/TextField.tsx#L28C18-L28C18